src/components/timecard/add-timecard-period/AddTimeCardPeriod.jsx

```diff
@@ -1,14 +1,12 @@
-import { useTimecardContext } from '../../../context/TimecardContext';
 import { Link } from 'react-router-dom';
 import { addTimePeriodHeading } from '../../../utils/time-entry-utils/timeEntryUtils';
+import { PropTypes } from 'prop-types';
 
-const AddTimeCardPeriod = () => {
-  const { setSummaryErrors, setNewTimeEntry } = useTimecardContext();
-
+const AddTimeCardPeriod = ({ setSummaryErrors, setAddNewTimeEntry }) => {
```

Comment:
I think this is in the naming again. I think the prop should be onAddNewTimeEntry which should just be called and the handler passed through would then decide what it wanted to do with that event.